tasks/task_irma.wdl

```diff
@@ -0,0 +1,98 @@
+version 1.0
+
+task irma {
+  input {
+    File read1
+    File read2
+    String samplename
+    Boolean keep_ref_deletions = true
+    String irma_module = "FLU"
+    String read_basename = basename(read1)
+    String docker = "quay.io/staphb/irma:1.0.3"
+	Int memory = 8
+	Int cpu = 4
+  }
+  command <<<
+    date | tee DATE
+    #capture reads as bash variables
+    read1=~{read1}
+    read2=~{read2}
+    # set cat command based on compression
+    if [[ "~{read1}" == *".gz" ]] ; then
+      cat_reads="zcat"
+    else
+      cat_reads="cat"
+    fi
+    # capture irma vesion
+    IRMA | head -n1 | awk -F' ' '{ print "IRMA " $5 }' | tee VERSION
+    # set config if needed
+    if ~{keep_ref_deletions}; then 
+      touch irma_config.sh
+      echo 'DEL_TYPE="NNN"' >> irma_config.sh
+      echo 'ALIGN_PROG="BLAT"' >> irma_config.sh
+    fi
+    # format reads, if needed
+    read_header=$(${cat_reads} ~{read1} | head -n1)
+    if ! [[ "${read_header}" =~ @(.+?)[_[:space:]][123]:.+ ]]; then
```

Comment:
maybe also a comment here that explains what IRMA considers a bad header vs a good header might be good for posterity

Reply:
Unfortunately, I don't have a great explanation myself, but the gist of it is that headers that don't fit that regex will lead to a failure. I can reach out to IRMA dev folks to see if they have more insight on the inclusion of this strict header check.